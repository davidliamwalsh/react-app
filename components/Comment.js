import { Component } from 'react'
import moment from 'moment'

class Comment extends Component {

  render () {

    const { comment } = this.props

    return  <div className="c-comments">
      <div className="c-comments__comment__card">
        <p className="c-comments__comment__card--name">Comment by {comment.name}</p>
        <p className="c-comments__comment__card--body">{comment.body}</p>
        <p className="c-comments__comment__card--date">Posted {moment(comment.createdAt).fromNow()}</p>
      </div>
    </div>
  }
}

export default Comment