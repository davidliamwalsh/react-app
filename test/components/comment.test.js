import Comment from '../../components/Comment'

describe('<Comment />', () => {
  it('it renders correctly', () => {
    const comment = {
      id: '1',
      name: 'David Walsh',
      body: 'This is the comment body',
      created_at: new Date(Date.UTC(97, 1, 2, 3, 4, 5))
    }

    const component = shallow(
      <Comment comment={comment} />
    )

    expect(component).toMatchSnapshot()
  })
}) 