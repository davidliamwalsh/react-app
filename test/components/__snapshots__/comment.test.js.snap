// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Comment /> it renders correctly 1`] = `ShallowWrapper {}`;